django/contrib/auth/middleware.py

```diff
@@ -21,7 +20,6 @@ def process_request(self, request):
             "'django.contrib.sessions.middleware.SessionMiddleware' before "
```

Comment:
It seems like this check should be moved to `contrib.auth.get_user()` and this empty middleware should be deprecated?

Also, the `if not hasattr(request, 'user')` message in `RemoteUserMiddleware` must be updated as `AuthenticationMiddleware` is no longer required. Other documentation changes are also needed. This could be backwards incompatible if a project isn't using `AuthConfig`, likely not a huge concern but should be mentioned in the release notes.